presto-main/src/main/java/com/facebook/presto/sql/planner/plan/DistinctLimitNode.java

```diff
@@ -74,18 +85,39 @@ public long getLimit()
         return hashSymbol;
     }
 
+    @JsonProperty("distinctSymbols")
     public List<Symbol> getDistinctSymbols()
     {
         if (hashSymbol.isPresent()) {
-            return ImmutableList.copyOf(Iterables.filter(getOutputSymbols(), not(hashSymbol.get()::equals)));
+            return ImmutableList.copyOf(Iterables.filter(distinctSymbols, not(hashSymbol.get()::equals)));
         }
-        return getOutputSymbols();
+        return distinctSymbols;
+    }
+
+    @JsonProperty("nonDistinctSymbols")
```

Comment:
I think the only thing this class needs is an explicit list of distinctSymbols. I think it would be much simpler if the behavior here was that the DistinctLimit would do a distinct on the explicit columns and pass through all the other columns (which is what the underlying operator does).